pkgs/by-name/op/opcua-commander/package.nix

```
@@ -0,0 +1,46 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+, typescript
+, esbuild
+, makeWrapper
+, nodejs
+}:
+buildNpmPackage rec {
+  pname = "opcua-commander";
+  version = "0.37.0";
+
+  src = fetchFromGitHub {
+    owner = "node-opcua";
+    repo = "opcua-commander";
+    rev = version;
+    hash = "sha256-wQXSSNinY85Ti+D/zklYP2N8IP3OsN9xQNJuuQr4kVU=";
+  };
+
+  npmDepsHash = "sha256-Ux1X/3sam9WHrTfqoWv1r9p3pJOs6BaeFsxHizAvjXA=";
+  nativeBuildInputs = [ esbuild typescript makeWrapper ];
+
+  buildPhase = ''
+    tsc -b && npx -y ${esbuild}/bin/esbuild \
+      --bundle --outfile=bin/bundle.js \
+      --external:blessed \
+      --external:chalk \
+      --external:wrap-ansi \
+      --minify-syntax \
+      --sourcemap \
+      --platform=node lib/index.ts
+  '';
```

We can even drop `buildPhase` and let default one do its job:
```suggestion
  postPatch = ''
    substituteInPlace package.json \
      --replace-warn "npx -y esbuild" "npx -y ${esbuild}/bin/esbuild"
  '';
```

---

nice! I'll keep this one in mind for future packages, thanks!